tests/vmi_networking_test.go

```diff
@@ -118,300 +118,319 @@ var _ = Describe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:comp
+				}
 
-			By("checking k6t-eth0 MTU inside the pod")
-			vmiPod := tests.GetRunningPodByVirtualMachineInstance(outboundVMI, tests.NamespaceTestDefault)
-			output, err := tests.ExecuteCommandOnPod(
-				virtClient,
-				vmiPod,
-				"compute",
-				[]string{"cat", "/sys/class/net/k6t-eth0/mtu"},
-			)
-			log.Log.Infof("k6t-eth0 mtu is %v", output)
-			Expect(err).ToNot(HaveOccurred())
+				switch destination {
+				case "Internet":
+					if netutils.IsIPv6String(inboundVMI.Status.Interfaces[0].IP) {
+						By("Skipping external connectivity check on IPv6 cluster")
+						return
+					}
+					addr = "kubevirt.io"
+				case "InboundVMI":
+					addr = inboundVMI.Status.Interfaces[0].IP
+				case "InboundVMIWithPodNetworkSet":
+					addr = inboundVMIWithPodNetworkSet.Status.Interfaces[0].IP
+				case "InboundVMIWithCustomMacAddress":
+					addr = inboundVMIWithCustomMacAddress.Status.Interfaces[0].IP
+				}
 
-			output = strings.TrimSuffix(output, "\n")
-			mtu, err := strconv.Atoi(output)
-			Expect(err).ToNot(HaveOccurred())
+				payloadSize := 0
+				ipHeaderSize := 28 // IPv4 specific
+
+				By("checking k6t-eth0 MTU inside the pod")
+				vmiPod := tests.GetRunningPodByVirtualMachineInstance(outboundVMI, tests.NamespaceTestDefault)
+				output, err := tests.ExecuteCommandOnPod(
+					virtClient,
+					vmiPod,
+					"compute",
+					[]string{"cat", "/sys/class/net/k6t-eth0/mtu"},
+				)
+				log.Log.Infof("k6t-eth0 mtu is %v", output)
+				Expect(err).ToNot(HaveOccurred())
 
-			Expect(mtu > 1000).To(BeTrue())
+				output = strings.TrimSuffix(output, "\n")
+				mtu, err := strconv.Atoi(output)
+				Expect(err).ToNot(HaveOccurred())
 
-			payloadSize = mtu - ipHeaderSize
-			expectedMtuString := fmt.Sprintf("mtu %d", mtu)
+				Expect(mtu > 1000).To(BeTrue())
 
-			By("checking eth0 MTU inside the VirtualMachineInstance")
-			expecter, err := tests.LoggedInCirrosExpecter(outboundVMI)
-			Expect(err).ToNot(HaveOccurred())
-			defer expecter.Close()
+				payloadSize = mtu - ipHeaderSize
+				expectedMtuString := fmt.Sprintf("mtu %d", mtu)
 
-			addrShow = "ip address show eth0\n"
-			resp, err := expecter.ExpectBatch([]expect.Batcher{
-				&expect.BSnd{S: "\n"},
-				&expect.BExp{R: "\\$ "},
-				&expect.BSnd{S: addrShow},
-				&expect.BExp{R: fmt.Sprintf(".*%s.*\n", expectedMtuString)},
-				&expect.BSnd{S: "echo $?\n"},
-				&expect.BExp{R: "0"},
-			}, 180*time.Second)
-			log.Log.Infof("%v", resp)
-			Expect(err).ToNot(HaveOccurred())
+				By("checking eth0 MTU inside the VirtualMachineInstance")
+				expecter, err := tests.LoggedInCirrosExpecter(outboundVMI)
+				Expect(err).ToNot(HaveOccurred())
+				defer expecter.Close()
+
+				addrShow = "ip address show eth0\n"
+				resp, err := expecter.ExpectBatch([]expect.Batcher{
+					&expect.BSnd{S: "\n"},
+					&expect.BExp{R: "\\$ "},
+					&expect.BSnd{S: addrShow},
+					&expect.BExp{R: fmt.Sprintf(".*%s.*\n", expectedMtuString)},
+					&expect.BSnd{S: "echo $?\n"},
+					&expect.BExp{R: "0"},
```

Comment:
you might want to use \n0 as #3543 fixed,
Alona would create a general mechanisem, and not sure here here there is an issue, but just up to you meanwhile

same for the 2nd batch, which is bit more sensitive, i am not sure the error can happen there, but better be safe,
because we did see some lines which are prone for problems like

```
		&expect.BSnd{S: "sudo ip -6 route add default via fd10:0:2::1 src fd10:0:2::2\n"},
		&expect.BExp{R: prompt},
		&expect.BSnd{S: "echo $?\n"},
		&expect.BExp{R: "0"},
```
where the 0 of the first input was catched as the 2 EXP (in case that i am not wrong and this was the atomic block that caused it without other code around that was affecting it)

Reply:
Tha is not related to this PR. I am only using existing tests with different environment setup.

Reply:
> That is not related to this PR. I am only using existing tests with different environment setup.
> 
> Or do you suspect it is related to IPv6 fails?

not that i know of, i just say that if there is flakiness , or will be flakiness, this is one thing i would surely check